client/dashboard/utils/domain.ts

```diff
@@ -98,7 +98,7 @@ const shouldUpgradeToMakeDomainPrimary = ( {
 	user: User;
 } ) => {
 	return (
-		( isRegisteredDomain( domain ) || domain.type === DomainTypes.MAPPED ) &&
+		domain.subtype.id === DomainSubtype.DOMAIN_CONNECTION &&
```

Comment:
```suggestion
		[ DomainSubtype.DOMAIN_CONNECTION, DomainSubtype.DOMAIN_REGISTRATION ].includes(
			domain.subtype.id
		) &&
```

Reply:
And I think we should also change the `isRegisteredDomain` function to check if the subtype is `DOMAIN_REGISTRATION`